src/js/emscripten-settings.ts

```diff
@@ -110,6 +110,15 @@ function mountLocalDirectories(mounts: string[]): PreRunFunc {
   };
 }
 
+function computeVersionTuple(Module: Module): [number, number, number] {
+  const versionHex = Module.HEAPU32[Module._Py_Version / 4].toString(16);
+  const res = versionHex
+    .padStart(8, "0")
+    .match(/.{1,2}/g)!
+    .map((x) => parseInt(x, 16));
+  return res as any;
+}
+
```

Comment:
```suggestion
function computeVersionTuple(Module: Module): [number, number, number] {
 const versionInt = Module.HEAPU32[Module._Py_Version >>> 2];
  const major = (versionInt >>> 24) & 0xFF;
  const minor = (versionInt >>> 16) & 0xFF;
  const micro = (versionInt >>> 8) & 0xFF;
  return [major, minor, micro];
}
```



Question: would an unsigned right shift be better? We are converting a number → hex string, then padding it, splitting it, and parsing it back to numbers (which might be unnecessary conversions). Please feel more than free to correct me if I am wrong somewhere :)

Reply:
Your code makes sense to me, let's try it.

Reply:
@agriyakhetarpal do you want to send this change as a pr to python or should I do it?

Reply:
I would love to, thanks!